paddle/fluid/operators/mkldnn/lrn_mkldnn_op.cc

```diff
@@ -76,66 +52,41 @@ class LRNMKLDNNOpKernel : public paddle::framework::OpKernel<T> {
-
-    auto forward_desc = mkldnn::lrn_forward::desc{mkldnn::prop_kind::forward,
-                                                  mkldnn::lrn_across_channels,
-                                                  src_md,
-                                                  n,
-                                                  alpha,
-                                                  beta,
-                                                  k};
-
-    auto src_memory_pd = mkldnn::memory::primitive_desc{src_md, mkldnn_engine};
-
-    if (!is_test) {
-      const std::string key = ctx.op().Output("Out");
-      const std::string key_src_memory = key + "@lrn_src_memory";
-      const std::string key_pd = key + "@lrn_pd";
-      const std::string key_workspace_memory = key + "@lrn_workspace_memory";
-
-      auto forward_pd = insert_to_context<mkldnn::lrn_forward::primitive_desc>(
-          key_pd, dev_ctx, forward_desc, mkldnn_engine);
-
-      auto src_memory = insert_to_context<mkldnn::memory>(
-          key_src_memory, dev_ctx, src_memory_pd);
-
-      src_memory->set_data_handle(
-          static_cast<void*>(const_cast<T*>(input_data)));
-
-      auto dst_memory = mkldnn::memory(forward_pd->dst_primitive_desc(),
-                                       static_cast<void*>(output_data));
-      auto workspace_memory = insert_to_context<mkldnn::memory>(
-          key_workspace_memory, dev_ctx,
-          forward_pd->workspace_primitive_desc());
-
-      run_primitive(*forward_pd, *src_memory, *workspace_memory, dst_memory);
-
-      out->set_layout(framework::DataLayout::kMKLDNN);
-      out->set_format(platform::GetMKLDNNFormat(dst_memory));
-    } else {
-      auto forward_pd =
-          mkldnn::lrn_forward::primitive_desc{forward_desc, mkldnn_engine};
-      auto src_memory = mkldnn::memory{
-          src_memory_pd, static_cast<void*>(const_cast<T*>(input_data))};
-      auto workspace_memory =
-          mkldnn::memory{forward_pd.workspace_primitive_desc()};
-      auto dst_memory = mkldnn::memory(forward_pd.dst_primitive_desc(),
-                                       static_cast<void*>(output_data));
-
-      run_primitive(forward_pd, src_memory, workspace_memory, dst_memory);
-
-      out->set_layout(framework::DataLayout::kMKLDNN);
-      out->set_format(platform::GetMKLDNNFormat(dst_memory));
-    }
+    // Format and dims are assumed to be the same for dst and src
+    auto md = paddle::platform::MKLDNNMemDesc(
+        dims, platform::MKLDNNGetDataType<T>(), x->format());
+
+    const std::string key = platform::LRNMKLDNNHandler::GetHash(
+        dims, n, alpha, beta, k, x->format(), ctx.op().Output("Out"));
+
+    platform::LRNMKLDNNHandler handler(ctx.Attr<bool>("is_test"), dev_ctx,
+                                       mkldnn_engine, key);
+    auto src_memory =
+        handler.AcquireSrcMemory(md, platform::to_void_cast<T>(input_data));
+
+    auto lrn_pd = handler.AcquireLRNPrimitiveDescriptor(md, n, alpha, beta, k);
```

Comment:
Since AcquireLRNPrimitiveDescriptor needs to internally create  a primitive descriptor this method needs to be called, but there is no need to save the result, as it is nowhere used.

Reply:
Ok. Good catch. Ultimately I will try not too expose AcquirePD to op. I have added it to my TODO list in refactoring tracking issue (#18531).